lighthouse-core/computed/computed-artifact.js

```diff
@@ -12,9 +12,11 @@ const log = require('lighthouse-logger');
  * Decorate computableArtifact with a caching `request()` method which will
  * automatically call `computableArtifact.compute_()` under the hood.
  * @template {{name: string, compute_(artifacts: unknown, context: LH.Artifacts.ComputedContext): Promise<unknown>}} C
+ * @template {Array<keyof FirstParamType<C['compute_']>>} K
  * @param {C} computableArtifact
+ * @param {(K & ([keyof FirstParamType<C['compute_']>] extends [K[number]] ? unknown : never)) | null} keys
  */
-function makeComputedArtifact(computableArtifact) {
+function makeComputedArtifact(computableArtifact, keys) {
```

Comment:
add documentation for `keys` as well? Unclear from just having the parameter why it's needed